schema-registry/src/test/java/io/streamnative/pulsar/handlers/kop/schemaregistry/resources/SchemaResourceTest.java

```diff
@@ -0,0 +1,527 @@
+                "/subjects/Kafka-value/versions",
+                "{\"schema\": \"{\\\"type\\\": \\\"string\\\"}\"}",
+                "application/vnd.schemaregistry.v1+json"));
+
+        assertEquals("{\n"
+                + "  \"id\" : 2,\n"
+                + "  \"version\" : 2\n"
+                + "}", server.executePost(
+                "/subjects/Kafka-value",
+                "{\"schema\": \"{\\\"type\\\": \\\"int\\\"}\"}",
+                "application/vnd.schemaregistry.v1+json"));
+    }
+
+    @Test
+    public void checkCompatibility() throws Exception {
+
+        // set FULL_TRANSITIVE
+        assertEquals("{\n"
+                + "  \"compatibilityLevel\" : \"" + CompatibilityChecker.Mode.FULL_TRANSITIVE + "\"\n"
+                + "}", server.executeMethod("/config/Kafka-value",
+                "{\n"
+                        + "  \"compatibility\" : \"" + CompatibilityChecker.Mode.FULL_TRANSITIVE + "\"\n"
+                        + "}", "PUT", "application/json",
+                null
+        ));
+
+        assertEquals("{\n"
+                + "  \"id\" : 1\n"
+                + "}", server.executePost(
+                "/subjects/Kafka-value/versions",
+                "{\"schema\": \"{\\\"type\\\": \\\"string\\\"}\"}",
+                "application/vnd.schemaregistry.v1+json"));
+
+        // verify the same schema is compatible
+        assertEquals("{\n"
+                + "  \"is_compatible\" : true\n"
+                + "}", server.executePost(
+                "/compatibility/subjects/Kafka-value/versions/latest",
+                "{\"schema\": \"{\\\"type\\\": \\\"string\\\"}\"}",
+                "application/vnd.schemaregistry.v1+json"));
+
+        // verify a non-compatible schema
+        assertEquals("{\n"
+                + "  \"is_compatible\" : false\n"
+                + "}", server.executePost(
+                "/compatibility/subjects/Kafka-value/versions/latest",
+                "{\"schema\": \"{\\\"type\\\": \\\"long\\\"}\"}",
+                "application/vnd.schemaregistry.v1+json"));
+
+        // set NONE
+        assertEquals("{\n"
+                + "  \"compatibilityLevel\" : \"" + CompatibilityChecker.Mode.NONE + "\"\n"
+                + "}", server.executeMethod("/config/Kafka-value",
+                "{\n"
+                        + "  \"compatibility\" : \"" + CompatibilityChecker.Mode.NONE + "\"\n"
+                        + "}", "PUT", "application/json",
+                null
+        ));
+
+        // verify a non-compatible schema, now the result is 'true'
+        assertEquals("{\n"
+                + "  \"is_compatible\" : true\n"
+                + "}", server.executePost(
+                "/compatibility/subjects/Kafka-value/versions/latest",
+                "{\"schema\": \"{\\\"type\\\": \\\"long\\\"}\"}",
+                "application/vnd.schemaregistry.v1+json"));
+    }
+
+    @Test
+    public void createSchemaWithCheckCompatibility() throws Exception {
```

Comment:
Could you add tests for other compatibility strategies? We can add them in another PR.